camayoc/api.py

```diff
@@ -121,30 +126,70 @@ def __init__(self, response_handler=None, url=None):
         else:
             self.response_handler = response_handler
 
+        if self.authenticate:
+            self.login()
+
+    def login(self):
+        """Login to the server to receive an authorization token."""
+        self.authenticate = True
+        cfg = config.get_config().get('qcs', {})
+        server_username = cfg.get('username', 'admin')
+        server_password = cfg.get('password', 'pass')
+        login_request = self.request(
+            'POST',
+            urljoin(self.url, QCS_LOGIN_PATH),
+            json={
+                'username': server_username,
+                'password': server_password
+            }
+        )
+        self.token = login_request.json()['token']
+        return login_request
+
+    def logout(self):
+        """Start sending unauthorized requests.
+
+        There is no API interaction that need occur to logout.
+        We simply must send unauthorized requests.
+        """
+        self.authenticate = False
+        self.token = ''
+
+    def auth_header(self):
+        """Build the authorization header."""
+        if self.authenticate:
+            return {'Authorization': 'Token {}'.format(self.token)}
+        return {}
+
     def delete(self, endpoint, **kwargs):
         """Send an HTTP DELETE request."""
         url = urljoin(self.url, endpoint)
-        return self.request('DELETE', url, **kwargs)
+        hdr = self.auth_header()
+        return self.request('DELETE', url, headers=hdr, **kwargs)
 
     def get(self, endpoint, **kwargs):
         """Send an HTTP GET request."""
         url = urljoin(self.url, endpoint)
-        return self.request('GET', url, **kwargs)
+        hdr = self.auth_header()
+        return self.request('GET', url, headers=hdr, **kwargs)
```

Comment:
hdr variable is being used only once, I think you can call the method directly here `return self.request('GET', url, headers=self.auth_header(), **kwargs)` or `return self.request('GET', url, headers=self.*_headers(), **kwargs)` if you change the header method signature

Reply:
Good call